fhir-server/src/main/java/com/ibm/fhir/server/rest/FHIRRestInteractionVisitorOffload.java

```diff
@@ -0,0 +1,246 @@
+        boolean skippableUpdate, String localIdentifier, List<Issue> warnings, boolean isDeleted, Integer ifNoneMatch, PayloadPersistenceResponse offloadResponse) throws Exception {
+
+        // Use doOperation for common exception handling
+        return doOperation(entryIndex, requestDescription, accumulatedTime, () -> {
+            // Try offloading storage of the payload. The offloadResponse will be null if not supported
+            String resourcePayloadKey = UUID.randomUUID().toString();
+            int newVersionNumber = Integer.parseInt(resource.getMeta().getVersionId().getValue());
+            PayloadPersistenceResponse actualOffloadResponse = storePayload(resource, resource.getId(), newVersionNumber, resourcePayloadKey);
+
+            // Resource doesn't change, so no need to return it
+            FHIRRestOperationResponse result = new FHIRRestOperationResponse(null, null, actualOffloadResponse);
+            result.setDeleted(isDeleted);
+            
+            return result;
+        });
+    }
+
+    @Override
+    public FHIRRestOperationResponse doPatch(int entryIndex, FHIRPersistenceEvent event, Entry validationResponseEntry, String requestDescription, FHIRUrlParser requestURL, long accumulatedTime,
+        String type, String id, Resource resource, Resource prevResource, FHIRPatch patch, String ifMatchValue, String searchQueryString,
+        boolean skippableUpdate, List<Issue> warnings, String localIdentifier, PayloadPersistenceResponse offloadResponse) throws Exception {
+        // Use doOperation for common exception handling
+        return doOperation(entryIndex, requestDescription, accumulatedTime, () -> {
+
+            // Try offloading storage of the payload. The offloadResponse will be null if not supported
+            String resourcePayloadKey = UUID.randomUUID().toString();
+            int newVersionNumber = Integer.parseInt(resource.getMeta().getVersionId().getValue());
+            PayloadPersistenceResponse actualOffloadResponse = storePayload(resource, resource.getId(), newVersionNumber, resourcePayloadKey);
+
+            // Resource doesn't change, so no need to return it
+            return new FHIRRestOperationResponse(null, null, actualOffloadResponse);
+        });
+    }
+
+    @Override
+    public FHIRRestOperationResponse doInvoke(String method, int entryIndex, Entry validationResponseEntry, String requestDescription, FHIRUrlParser requestURL, long accumulatedTime, FHIROperationContext operationContext, String resourceTypeName, String logicalId,
+            String versionId, Resource resource, MultivaluedMap<String, String> queryParameters) throws Exception {
+        // NOP
+        return null;
+    }
+
+    @Override
+    public FHIRRestOperationResponse doDelete(int entryIndex, String requestDescription, FHIRUrlParser requestURL, long accumulatedTime, String type, String id, String searchQueryString) throws Exception {
+        // NOP
+        return null;
```

Comment:
When I was implementing the offload I was working through avoiding storing the deletion marker payload because now we have better handling for return null resources in the persistence layer read methods, but I think there are more use-cases to cover. I think for now I need to store the payload and then we can deal with skipping the store as a later optimization.

Reply:
payload is no longer stored for the delete case, so this is resolved